### PR TITLE
Fix Potential Missed Serializable Transaction Conflicts

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- Catch transaction conflicts

**Implementation Description (bullets)**:
- Don't read at commitTs + 1 but instead at commitTs
- Tests

**Testing (What was existing testing like?  What have you done to improve it?)**:
- Added a test that exposes the bug

**Concerns (what feedback would you like?)**:
- Is this actually safe

**Where should we start reviewing?**: SerializableTransaction

**Priority (whenever / two weeks / yesterday)**: five years ago 🔥

### CODE REVIEW

File: atlasdb-processors/build.gradle [hunk 1]
```diff
@@ -7,8 +7,8 @@ dependencies {
     compileOnly group: 'com.google.auto.service', name: 'auto-service'
 
     compile group: 'com.google.guava', name: 'guava'
-    implementation group: 'com.squareup', name: 'javapoet'
-    implementation group: 'com.palantir.goethe', name: 'goethe'
+    compile group: 'com.squareup', name: 'javapoet'
+    compile group: 'com.palantir.goethe', name: 'goethe'
```
Comment (Jolyon-S) [inline atlasdb-processors/build.gradle:11]: do we want to commit this change?

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java [hunk 2]
```diff
@@ -1408,6 +1414,47 @@ public void testGetSortedColumnsConflictForReadBoundaryEntry() {
...
+
+        assertThatThrownBy(() -> manager.runTaskThrowOnConflict(tx -> {
+            long myTs = tx.getTimestamp();
+            tx.get(TEST_TABLE_SERIALIZABLE, ImmutableSet.of(CELL_ONE));
+
+            // This write is done to ensure that the transaction actually has writes that need to commit
+            tx.put(TEST_TABLE_SERIALIZABLE, ImmutableMap.of(CELL_TWO, BYTES_TWO));
+
+            keyValueService.put(TEST_TABLE_SERIALIZABLE, ImmutableMap.of(CELL_ONE, BYTES_TWO), myTs + 1);
+            transactionService.putUnlessExists(myTs + 1, myTs + 2);
```
Comment (Jolyon-S) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:1434]: we should comment that this is effectively orchestrating a "B" transaction (if this is "ABA").

File: atlasdb-processors/build.gradle (hunk 1 above)
Comment (gmaretic) [reply atlasdb-processors/build.gradle:11]: 🔥

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java (hunk 2 above)
Comment (gmaretic) [reply atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractSerializableTransactionTest.java:1434]: Yeah maybe a bit more comment on what we are doing

**Review by gmaretic — APPROVED** [review]
Safe as far as I can tell

File: atlasdb-processors/build.gradle (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-processors/build.gradle:11]: Nope, will remove! (At least for now)

**Pull request merged**